### PR TITLE
Add public interface for natural_loopst and natural_loopt

### PR DESCRIPTION
**Pull request by smowton**
Makes it easier to alter the implementation of these types. No behavioural change intended. Depends on #5051; do not review the first two commits.

### CODE REVIEW

File: src/analyses/natural_loops.h [hunk 1]
```diff
@@ -28,6 +28,79 @@ Author: Georg Weissenbacher, georg@weissenbacher.name
...
+
+/// A natural loop, specified as a set of instructions
+template <class P, class T>
+class natural_loop_templatet
+{
+  typedef natural_loops_templatet<P, T> natural_loopst;
+
+public:
+  typedef std::set<T> loop_instructionst;
+  loop_instructionst loop_instructions;
```
Comment (peterschrammel) [inline src/analyses/natural_loops.h:44]: Why is this public?

**Review by peterschrammel — COMMENTED** [review]
CI seems to fail

File: src/analyses/natural_loops.h (hunk 1 above)
Comment (smowton) [reply src/analyses/natural_loops.h:44]: Doh, fixed it now, and made these private as you say.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: ec02d1418b24357fdf8d679cc964e7d9c83fa513).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/124559280

**Comment by smowton** [comment]
@peterschrammel now passing, please re-review

**Review by kroening — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**